src/rpm_ostree/cli_status.rs

```diff
@@ -104,15 +104,30 @@ pub fn parse_booted_updates_stream(status: &StatusJson) -> Result<String> {
     Ok(json.base_metadata.stream)
 }
 
-/// Parse updates stream for pending deployment from status object.
-pub fn parse_pending_updates_stream(status: &StatusJson) -> Result<String> {
-    let pending_json = status.deployments[0].clone();
-    ensure!(!pending_json.booted, "no pending deployment found");
-    ensure!(
-        !pending_json.base_metadata.stream.is_empty(),
-        "empty stream value"
-    );
-    Ok(pending_json.base_metadata.stream)
+/// Parse pending deployment from status object.
+pub fn parse_pending_deployment(
+    status: &StatusJson,
+    preferred: &Release,
+) -> Result<(Release, String)> {
+    // Pick a deployment among all staged ones, if any. In case of
```

Comment:
There can only be one staged, FWIW.

Reply:
Thanks for highlighting that. I'll note it down in a comment, and simplify the code accordingly.